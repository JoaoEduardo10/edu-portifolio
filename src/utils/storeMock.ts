import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

const store = mockStore({
  menu: {
    menuOn: false,
    colorOn: true,
  },
  completeProject: {
    id: "1234",
    project: false,
  },
});

export { store };
