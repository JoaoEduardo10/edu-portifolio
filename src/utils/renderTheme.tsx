/* eslint-disable @typescript-eslint/no-explicit-any */
import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../style/themes";
import { Provider } from "react-redux";
import { GlobalStyles } from "../style/globalStyles";
import { ToolkitStore } from "@reduxjs/toolkit/dist/configureStore";
import { store } from "./storeMock";

const renderTheme = (children: React.ReactNode, storeParams?: ToolkitStore) => {
  const isStore = storeParams ?? store;
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={isStore}>
        {children}
        <GlobalStyles />
      </Provider>
    </ThemeProvider>
  );
};

export { renderTheme };
