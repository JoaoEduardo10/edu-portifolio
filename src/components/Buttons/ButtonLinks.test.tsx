import React from "react";
import { renderTheme } from "../../utils/renderTheme";
import { describe, it } from "vitest";
import { ButtonLinks } from ".";
import { screen } from "@testing-library/react";

describe("<ButtonLinks />", () => {
  it("should a render ButtoDownload component", () => {
    renderTheme(<ButtonLinks link="https://test" text="test" />);

    const button = screen.getByRole("button");
    const link = screen.getByRole("link", { name: "test" });

    expect(button).toBeInTheDocument();
    expect(link).toHaveAttribute("href", "https://test");
  });

  it("should to match snapshot", () => {
    renderTheme(<ButtonLinks link="https://test" text="test" />);

    const button = screen.getByRole("button");

    expect(button).toMatchSnapshot();
  });
});
