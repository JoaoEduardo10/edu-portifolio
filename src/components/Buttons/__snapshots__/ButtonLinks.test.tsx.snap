// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ButtonLinks /> > should to match snapshot 1`] = `
.c0 {
  padding: 1rem 2.5rem;
  border: none;
  font-size: 2rem;
  background-image: linear-gradient(to left,#A115E4,#8257E6);
  border-radius: 2rem;
  margin-top: 1rem;
}

.c0 a {
  color: #fff;
}

.c0:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

@media (max-width:950px) {
  .c0:hover {
    -webkit-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }
}

<button
  class="c0"
>
  <a
    href="https://test"
    rel="noreferrer"
    target="blank"
  >
    test
  </a>
</button>
`;
